app.py

```diff
@@ -1,13 +1,101 @@
-from flask import Flask, render_template
+from flask import Flask, session, redirect, render_template, flash
+from flask_login import LoginManager, UserMixin, login_user, logout_user, login_required
+from forms import LoginForm, SignUpForm
+import bcrypt
+import csv
 
 app = Flask(__name__)
 app.secret_key = 'secret_key'
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view = 'login'
+# allow the use of @login_required on endpoints which require an account
+# unauthenticated users will be redirected to login page
+
+class User(UserMixin):
+    def __init__(self, username, email, first_name, last_name, password=None):
+        self.id = username
+        self.first_name = first_name
+        self.last_name = last_name
+        self.email = email
+        self.password = password
+
+    def __repr__(self):
+        return 'User({})'.format(self.id)
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    """retrieve a user object for the current user while hiding password"""
+    user = find_user(user_id)
+    if user:
+        user.password = None
+    return user
+
+
+def find_user(username):
+    """ TODO : this will later be changed to searching the database"""
```

Comment:
You can use the code template in follower.py if you need to find a user in any of the databases.